src/graph/graph-state-manager.spec.ts

```diff
@@ -0,0 +1,150 @@
+require('dotenv').config({ path: '.env.test' });
+
+import { Test, TestingModule } from '@nestjs/testing';
+import { GraphStateManager } from './graph-state-manager';
+import { Action, ConnectAction, Connection, DsnpKeys, EnvironmentType, GraphKeyPair, ImportBundle, PageData } from '@dsnp/graph-sdk';
+import { ConfigService } from '../config/config.service';
+import { configModuleOptions } from '../config/env.config';
+import { ConfigModule } from '@nestjs/config';
+import { GraphManagerModule } from './graph-state.module';
+
+type ProcessEnv = {
+  REDIS_URL: string;
+  FREQUENCY_URL: string;
+  PROVIDER_ID: string;
+  PROVIDER_BASE_URL: string;
+  PROVIDER_USER_GRAPH_ENDPOINT: string;
+  PROVIDER_ACCESS_TOKEN: string;
+  BLOCKCHAIN_SCAN_INTERVAL_MINUTES: string;
+  QUEUE_HIGH_WATER: string;
+  GRAPH_ENVIRONMENT_TYPE: string;
+  GRAPH_ENVIRONMENT_CONFIG: string;
+};
+
+describe('GraphStateManager', () => {
+  const REDIS_URL = 'redis://localhost:6389';
+  const FREQUENCY_URL = 'ws://localhost:9933';
+  const PROVIDER_ID = '1';
+  const PROVIDER_BASE_URL = 'https://some-provider';
+  const PROVIDER_USER_GRAPH_ENDPOINT = 'user-graph';
+  const PROVIDER_ACCESS_TOKEN = 'some-token';
+  const BLOCKCHAIN_SCAN_INTERVAL_MINUTES = '60';
+  const QUEUE_HIGH_WATER = '1000';
+  const GRAPH_ENVIRONMENT_TYPE = 'Mainnet';
+  const GRAPH_ENVIRONMENT_CONFIG = '{}';
+
+  const ALL_ENV = {
+    REDIS_URL,
+    FREQUENCY_URL,
+    PROVIDER_ID,
+    PROVIDER_BASE_URL,
+    PROVIDER_USER_GRAPH_ENDPOINT,
+    PROVIDER_ACCESS_TOKEN,
+    BLOCKCHAIN_SCAN_INTERVAL_MINUTES,
+    QUEUE_HIGH_WATER,
+    GRAPH_ENVIRONMENT_TYPE,
+    GRAPH_ENVIRONMENT_CONFIG,
+  };
+  let graphStateManager: GraphStateManager;
+  
+  beforeEach(async () => {
```

Comment:
For efficiency, this can probably be `beforeAll`--I think it likely only needs to be done once, no? If we need to reset state after each test you can do an `afterEach` hook that clears the existing GraphStates without having to re-setup the entire app module under test.

Reply:
Sure! just did

Reply:
Since a finalizer is implemented for `GraphStateManager`, we would not need to free state explicitly

Reply:
I thought the finalizer didn't work?

Reply:
its at the module level, when `GraphStateManager` is garbage collected it calls the free on the state from sdk

Reply:
the issue in node bridge is Graph class has not object to point to, since state is managed at rust side and all we have is a handle number, but in this case it works as when GraphStateManager is garbage collected it triggers the free state on finality

Reply:
ah, ok